### ISSUE TITLE
jib-core: incompatible with jackson 2.12.0 (Java 8 date/time type `java.time.Instant` not supported by default)

### ISSUE THREAD

**Issue by iamdanfox** [GoogleContainerTools/jib#2931]
**Environment**:

- *Jib version:* jib-core 0.16.0
- *Build tool:* Gradle 6.7.1
- *OS:* Mac, Linux

**Description of the issue**:

I'm writing a gradle plugin leveraging jib-core 0.16.0, and while it works fine on most projects, I've found that if gradle's plugin classpath happens to contain a newer version of jackon (specifically 2.12.0), then `Cache#retrieve` throws:

```
org.gradle.internal.UncheckedException: java.util.concurrent.ExecutionException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Instant` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: java.util.ArrayList[0]->com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate["sourceModificationTime"])
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:67)
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:41)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:107)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:555)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	...
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.util.concurrent.ExecutionException: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Instant` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: java.util.ArrayList[0]->com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate["sourceModificationTime"])
	at com.google.common.util.concurrent.AbstractFuture.getDoneValue(AbstractFuture.java:552)
	at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:513)
	at com.google.common.util.concurrent.FluentFuture$TrustedFuture.get(FluentFuture.java:82)
	at com.google.cloud.tools.jib.builder.steps.StepsRunner.realizeFutures(StepsRunner.java:94)
	at com.google.cloud.tools.jib.builder.steps.StepsRunner.lambda$buildImage$8(StepsRunner.java:346)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	... 1 more
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Instant` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: java.util.ArrayList[0]->com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate["sourceModificationTime"])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1276)
	at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:728)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4485)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3699)
	at com.google.cloud.tools.jib.json.JsonTemplateMapper.writeTo(JsonTemplateMapper.java:187)
	at com.google.cloud.tools.jib.json.JsonTemplateMapper.writeTo(JsonTemplateMapper.java:173)
	at com.google.cloud.tools.jib.hash.Digests.lambda$computeJsonDigest$0(Digests.java:44)
	at com.google.cloud.tools.jib.hash.Digests.computeDigest(Digests.java:102)
	at com.google.cloud.tools.jib.hash.Digests.computeJsonDigest(Digests.java:45)
	at com.google.cloud.tools.jib.cache.LayerEntriesSelector.generateSelector(LayerEntriesSelector.java:169)
	at com.google.cloud.tools.jib.cache.Cache.retrieve(Cache.java:182)
	at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:102)
	at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:38)
	... 6 more
```

It seems like this behaviour change landed in jackson in this commit: https://github.com/FasterXML/jackson-databind/commit/85c9c8544f0c4f01e88241acc1573746df4f755d#diff-4119ee3f674d1d2c64d16e0118a6cad1277bc30849b60970025af744453158e5R312-R320 and it seems the fix is pretty straightforward, `:jib-core` just needs to explicitly register a JavaTime module when setting up its ObjectMapper rather than relying on the old implicit behaviour to serialize & deserialize Instant types. I'm happy to submit a PR for this.

**Expected behavior**:

Ideally jib-core would work with a variety of different jackson versions, but most importantly latest (2.12.0 at time of writing).

**Steps to reproduce**:

  1. Clone GoogleContainerTools/jib
  2. update build.gradle to build against jackson 2.12.0
  3. Run `./gradlew :jib-core:test -i` and see existing tests fail, e.g.

```
com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStepTest > testRun FAILED
    com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Instant` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: java.util.ArrayList[0]->com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate["sourceModificationTime"])
        at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
        at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1276)
        at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:728)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
        at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
        at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4485)
        at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:3699)
        at com.google.cloud.tools.jib.json.JsonTemplateMapper.writeTo(JsonTemplateMapper.java:187)
        at com.google.cloud.tools.jib.json.JsonTemplateMapper.writeTo(JsonTemplateMapper.java:173)
        at com.google.cloud.tools.jib.hash.Digests.lambda$computeJsonDigest$0(Digests.java:44)
        at com.google.cloud.tools.jib.hash.Digests.computeDigest(Digests.java:102)
        at com.google.cloud.tools.jib.hash.Digests.computeJsonDigest(Digests.java:45)
        at com.google.cloud.tools.jib.cache.LayerEntriesSelector.generateSelector(LayerEntriesSelector.java:169)
        at com.google.cloud.tools.jib.cache.Cache.retrieve(Cache.java:206)
        at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:102)
        at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStepTest.buildFakeLayersToCache(BuildAndCacheApplicationLayerStepTest.java:139)
        at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStepTest.testRun(BuildAndCacheApplicationLayerStepTest.java:158)
```

**Comment by chanseokoh** [comment]
Actually, looks like https://github.com/FasterXML/jackson-databind/issues/2683 was the first issue to fail explicitly with `Instant` in 2.12.0. What you linked (https://github.com/FasterXML/jackson-databind/commit/85c9c8544f0c4f01e88241acc1573746df4f755d#) is the next one for yoda date/time types.

**Comment by loosebazooka** [comment]
@iamdanfox we released 2.7.1 with this fix. Thanks for the bug report and feedback.
